cmp/grid/columns/Column.js

```diff
@@ -239,31 +245,37 @@ export class Column {
                         {value, data: record} = agParams;
                     return elementRenderer(value, {record, agParams, column: me});
                 }
+
                 refresh() {return false}
             };
         }
 
-        // Support enhanced, absValue-aware sorting via GridSorters in GridModel.sortBy[].
         const sortCfg = find(gridModel.sortBy, {colId: ret.colId});
         if (sortCfg) {
             ret.sort = sortCfg.sort;
             ret.sortedAt = gridModel.sortBy.indexOf(sortCfg);
 
-            // ag-Grid sort impl. sources its primary values from the node's `groupData` property,
-            // which is not what we want when sorting treeColumns.
-            if (this.isTreeColumn) {
-                ret.comparator = (v1, v2, node1, node2) => {
-                    return sortCfg.comparator(node1.data[field], node2.data[field]);
-                };
-            } else {
-                ret.comparator = (v1, v2) => sortCfg.comparator(v1, v2);
-            }
-
+            // Delegate comparator sorting to absValue-aware GridSorters in GridModel.sortBy[].
+            ret.comparator = this.comparator;
         }
 
         // Finally, apply explicit app requests.  The customer is always right....
         return {...ret, ...this.agOptions};
     }
+
+    //--------------------
+    // Implementation
+    //--------------------
+    comparator = (v1, v2, node1, node2) => {
+        const sortCfg = find(this.gridModel.sortBy, {colId: this.colId}),
+            // ag-Grid sort impl. sources its primary values from the node's `groupData` property,
+            // which is not what we want when sorting treeColumns.
+            val1 = this.isTreeColumn ? node1.data[this.field] : v1,
+            val2 = this.isTreeColumn ? node2.data[this.field] : v2;
```

Comment:
Could we test if the tree-related switching here is still required? I think that ag-Grid made some updates in 19.1 to respect the valueGetter for sorting, which we install on line 209 above.